diff -r f3d6d92d3066 src/org/odk/collect/android/widgets/DateWidget.java
--- a/src/org/odk/collect/android/widgets/DateWidget.java	Tue May 08 10:08:29 2012 -0700
+++ b/src/org/odk/collect/android/widgets/DateWidget.java	Fri Jun 15 11:20:23 2012 +0530
@@ -21,9 +21,11 @@
 
 import android.content.Context;
 import android.view.Gravity;
+import android.view.View;
 import android.view.inputmethod.InputMethodManager;
 import android.widget.DatePicker;
 
+import java.lang.reflect.Field;
 import java.util.Calendar;
 import java.util.Date;
 
@@ -38,6 +40,8 @@
 
     private DatePicker mDatePicker;
     private DatePicker.OnDateChangedListener mDateListener;
+    private boolean hideDay = false;
+    private boolean hideMonth = false;
 
 
     public DateWidget(Context context, FormEntryPrompt prompt) {
@@ -46,7 +50,9 @@
         mDatePicker = new DatePicker(getContext());
         mDatePicker.setFocusable(!prompt.isReadOnly());
         mDatePicker.setEnabled(!prompt.isReadOnly());
-
+        
+        hideDayFieldIfNotInFormat(prompt);
+        
         mDateListener = new DatePicker.OnDateChangedListener() {
             @Override
             public void onDateChanged(DatePicker view, int year, int month, int day) {
@@ -82,6 +88,45 @@
         addView(mDatePicker);
     }
 
+    private void hideDayFieldIfNotInFormat(FormEntryPrompt prompt) {
+        String appearance = prompt.getQuestion().getAppearanceAttr();
+        if ( appearance == null ) return;
+        
+        if ( "month-year".equals(appearance) ) {
+        	hideDay = true;
+        } else if ( "year".equals(appearance) ) {
+        	hideMonth = true;
+        }
+
+        if ( hideMonth || hideDay ) {
+		    for (Field datePickerDialogField : this.mDatePicker.getClass().getDeclaredFields()) {
+		        if ("mDayPicker".equals(datePickerDialogField.getName()) ||
+		                "mDaySpinner".equals(datePickerDialogField.getName())) {
+		            datePickerDialogField.setAccessible(true);
+		            Object dayPicker = new Object();
+		            try {
+		                dayPicker = datePickerDialogField.get(this.mDatePicker);
+		            } catch (Exception e) {
+		                e.printStackTrace();
+		            }
+		            ((View) dayPicker).setVisibility(View.GONE);
+		        }
+		        if ( hideMonth ) {
+			        if ("mMonthPicker".equals(datePickerDialogField.getName()) ||
+			                "mMonthSpinner".equals(datePickerDialogField.getName())) {
+			            datePickerDialogField.setAccessible(true);
+			            Object monthPicker = new Object();
+			            try {
+			            	monthPicker = datePickerDialogField.get(this.mDatePicker);
+			            } catch (Exception e) {
+			                e.printStackTrace();
+			            }
+			            ((View) monthPicker).setVisibility(View.GONE);
+			        }
+		        }
+		    }
+        }
+    }
 
     private void setAnswer() {
 
@@ -110,10 +155,9 @@
 
     @Override
     public IAnswerData getAnswer() {
-        DateTime ldt =
-            new DateTime(mDatePicker.getYear(), mDatePicker.getMonth() + 1,
-                    mDatePicker.getDayOfMonth(), 0, 0);
-        // DateTime utc = ldt.withZone(DateTimeZone.forID("UTC"));
+        DateTime ldt = new DateTime(mDatePicker.getYear(), hideMonth ? 1 : mDatePicker.getMonth() + 1, 
+			(hideMonth || hideDay) ? 1 : mDatePicker.getDayOfMonth(), 0, 0);
+	// DateTime utc = ldt.withZone(DateTimeZone.forID("UTC"));
         return new DateData(ldt.toDate());
     }
 
